agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/telemetry/TelemetryClient.java

```diff
@@ -272,25 +273,24 @@ public BatchItemProcessor getStatsbeatBatchItemProcessor() {
     if (statsbeatBatchItemProcessor == null) {
       synchronized (batchItemProcessorInitLock) {
         if (statsbeatBatchItemProcessor == null) {
-          LocalStorageSystem localStorageSystem = null;
-          TelemetryPipelineListener telemetryPipelineListener = TelemetryPipelineListener.noop();
-          if (!readOnlyFileSystem) {
-            localStorageSystem =
-                new LocalStorageSystem(LocalStorageUtils.getOfflineStatsbeatFolder(), null);
-            telemetryPipelineListener = localStorageSystem.createTelemetryPipelineListener();
-          }
-
           HttpPipeline httpPipeline = LazyHttpClient.newHttpPipeLine(null);
           TelemetryPipeline telemetryPipeline =
               new TelemetryPipeline(httpPipeline, statsbeatConnectionString.getEndpoint());
 
+          TelemetryPipelineListener telemetryPipelineListener;
+          if (tempDir == null) {
+            telemetryPipelineListener = TelemetryPipelineListener.noop();
+          } else {
+            telemetryPipelineListener =
+                new LocalStorageTelemetryPipelineListener(
+                    TempDirs.getSubDir(tempDir, STATSBEAT_FOLDER_NAME),
+                    telemetryPipeline,
+                    LocalStorageStats.noop());
```

Comment:
nonessentialStatsbeat instead of noop?

Reply:
do we want to track Statsbeat failure read and write? i forgot if we discussed this before. please remind me again.

Reply:
we aren't tracking currently, so let's take this question separately from the PR

Reply:
sure, we need to fix retry count and exception count in that separate PR.